src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+            session._session._path_mapping_rules.sort(key=lambda rule: -len(rule.source_path.parts))
+
+            # =========================== TO BE DELETED ===========================
+            path_mapping_file_path: str = os.path.join(
+                session._session.working_directory, "path_mapping"
+            )
+            for rule in job_attachment_path_mappings:
+                rule["source_path"] = rule["destination_path"]
+
+            with open(path_mapping_file_path, "w", encoding="utf8") as f:
+                f.write(json.dumps([rule for rule in job_attachment_path_mappings]))
+            # =========================== TO BE DELETED ===========================
+
+            manifest_paths = session._asset_sync._check_and_write_local_manifests(
+                merged_manifests_by_root=merged_manifests_by_root,
+                manifest_write_dir=session._session.working_directory,
+            )
+
+            self.set_step_script(
+                manifests=manifest_paths,
+                path_mapping=path_mapping_file_path,
+                s3_settings=s3_settings,
+            )
+
+            session.run_task(
+                step_script=self._step_script,
+                task_parameter_values=dict[str, ParameterValue](),
+            )
+
+    def _vfs_handling(
+        self,
+        session: Session,
+        attachments: Attachments,
+        merged_manifests_by_root: dict[str, BaseAssetManifest],
+        s3_settings: JobAttachmentS3Settings,
+    ) -> bool:
+        fs_permission_settings: Optional[FileSystemPermissionSettings] = None
+        if session._os_user is not None:
+            if os.name == "posix":
+                if not isinstance(session._os_user, PosixSessionUser):
+                    raise ValueError(f"The user must be a posix-user. Got {type(session._os_user)}")
+                fs_permission_settings = PosixFileSystemPermissionSettings(
+                    os_user=session._os_user.user,
+                    os_group=session._os_user.group,
+                    dir_mode=0o20,
+                    file_mode=0o20,
+                )
+            else:
+                if not isinstance(session._os_user, WindowsSessionUser):
+                    raise ValueError(
+                        f"The user must be a windows-user. Got {type(session._os_user)}"
+                    )
+                if session.session.user is not None:
+                    fs_permission_settings = WindowsFileSystemPermissionSettings(
+                        os_user=session._os_user.user,
+                        dir_mode=WindowsPermissionEnum.WRITE,
+                        file_mode=WindowsPermissionEnum.WRITE,
+                    )
+
+        if (
+            attachments.fileSystem == JobAttachmentsFileSystem.VIRTUAL.value
+            and sys.platform != "win32"
+            and fs_permission_settings is not None
+            and os.environ is not None
+            and "AWS_PROFILE" in os.environ
```

Comment:
Is this condition necessary? I'm not sure if this is a requirement to launch the VFS.

Reply:
This is a move of existing code, I'm trying to keep it as-is since this part is not in scope of the change. I can probably remove some of the conditions, but that should be tested out with the VFS setup later.